fastcdc/fastcdc.h

```diff
@@ -240,180 +162,38 @@ class ChunkerTmpl {
     }
 
     // Init hash to all 1's to avoid zero-length chunks with min_size=0.
-    uint64_t hash = (uint64_t)-1;
-    // Skip the first min_size bytes, but "warm up" the rolling hash for 64
-    // rounds to make sure the 64-bit hash has gathered full "content history".
-    size_t i = cfg_.min_size > 64 ? cfg_.min_size - 64 : 0;
+    T hash = (T)-1;
+    // Skip the first min_size bytes, but "warm up" the rolling hash for enough
+    // rounds to make sure the hash has gathered full "content history".
+    size_t i = cfg_.min_size > khashbits_ ? cfg_.min_size - khashbits_ : 0;
     for (/*empty*/; i < cfg_.min_size; ++i) {
-      hash = (hash << 1) + gear[data[i]];
+      hash = ((hash << 1) + data[i]) * kmult;
     }
-    for (uint32_t stg = 0; stg < mask_stages && i < len; stg++) {
-      uint64_t mask = stages_[stg].mask;
-      size_t barrier = std::min(len, stages_[stg].barrier);
-      for (/*empty*/; i < barrier; ++i) {
-        if (!(hash & mask)) {
-          return i;
-        }
-        hash = (hash << 1) + gear[data[i]];
+    for (/*empty*/; i < len; ++i) {
+      if (hash <= kthreshold_) {
+        return i;
       }
+      hash = ((hash << 1) + data[i]) * kmult;
     }
     return i;
   }
 
+  static constexpr int khashbits_ = sizeof(T) * 8;
```

Comment:
Nit: Maybe use size_t instead of int as it's compared to a size_t above.

Reply:
Done.